libraries/AP_JSONParser/AP_JSONParser.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+/*
+  Convenience class for parsing json files.
+
+  The parsed data output consists of a stream of tokens which can be
+  retrieved from this object.
+
+  Once you have supplied a file to parse with parse_filepath, you
+  should be able to shift an object from your instance of this class.
+  The number returned (via parameter) is the number of key/value pairs
+  in the object.  Keys will always be strings.  Use shift_token_string
+  to get a string, then one of the shift_token_* methods to retrieve
+  the value (which may be another object with its count of key/value
+  pairs).
+
+  See tests directory for examples.
+
+ */
+
+#include "AP_JSONParser_config.h"
+#include <AP_Filesystem/AP_Filesystem_config.h>
+
+#if AP_JSONPARSER_ENABLED
+
+#include <AP_Math/vector3.h>
+
+// jsmn.h can't be included multiple times across a compilation as it
+// defines functions.  If they're static that's not a problem, but
+// they also aren't used in the .h file, so we need to declare them
+// static:
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+#define JSMN_STATIC
+#include "jsmn.h"
+#pragma GCC diagnostic pop
+
+#define AP_JSONPARSER_NUM_TOKENS 180  // allocated in cpp file
```

Comment:
static constexpr?

Reply:
Or, you know, just remove it as it is unused :-)